src/sources/kubernetes_logs/parser/cri.rs

```diff
@@ -107,49 +119,56 @@ impl FunctionTransform for Cri {
                         })
                     });
 
-                    let mut drop_original = true;
                     for (name, value) in captures {
-                        // If we're already overriding the original field, don't remove it after.
-                        if name == self.field {
-                            drop_original = false;
+                        match name.as_str() {
+                            MESSAGE_KEY => {
+                                // Insert either directly into `.` or `log_schema().message_key()`,
+                                // overwriting the original "full" CRI log that included additional fields.
+                                drop(log.insert(message_field, value));
+                            }
```

Comment:
Just having a little trouble comparing this to the previous logic.
We're now matching on specific values for `name` , but how do you know that `MESSAGE_KEY` needs "." for the `message_field` :thinking: ?

Wouldn't we need to match `name` to "."  to preserve previous behavior?

Reply:
Thanks for explaining this to me offline!